Automattic/jetpack
# Custom CSS: remove feature from Jetpack plugin

Fixes https://github.com/Automattic/vulcan/issues/447

## Proposed changes:

This PR removes the Jetpack Custom CSS feature from the Jetpack plugin, as WordPress has its own implementation.  

Simple and atomic sites should work as before. Self-hosted sites shouldn't show these three additional options in the `Additional CSS` section of the Customizer anymore:
<img width="200" alt="Screenshot 2024-08-21 at 10 39 25 AM" src="https://github.com/user-attachments/assets/f137a1a4-aed7-4d9b-8c35-06f81ddfc576">

In detail, this PR:
- Deletes all files in `modules/custom-css/`
- Deletes the `custom-css.php` module
- Removes references to these deleted files in phan, config, and test files
- Removes the code deprecation warning
- Removes the deprecation warnings we've previously added in the UI of the Customizer and Jetpack dashboard/settings
- Removes the Custom CSS section from the Jetpack Settings

### Other information:


## Jetpack product discussion


Project thread: pfwV0U-4M-p2

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






### Simple sites

#### Prerequisites
- Sync the Jetpack version from this branch with your sandbox, following one of these 2 methods:
  - Checkout the branch locally, then run `jetpack rsync jetpack` to start syncing the plugin with your sandbox
  - Run `bin/jetpack-downloader test jetpack remove/custom-css-warnings` directly from your sandbox

#### Testing
- Install a block-based theme on your sandbox test site, such as Twenty Twenty-Four
- Notice the `Appearance` menu item in the admin has a `Customize` subitem. Hit it.
- You should see an `Additional CSS` item in the Customizer. Hit it.
- You should see the 3 additional options mentioned above in the sidebar
- Write some custom CSS, .e.g.
```
body {
	background: purple;
}
```

- Save the changes and visit the frontend of your site
- You should see the changes you've made (note: the Start Fresh option doesn't work for block-based themes)
- Now, install a classic theme, such as Twenty Eleven
- Repeat the operations above:
  - In the `Appearance` menu, you should also see an `Additional CSS` subitem
  - The Start Fresh option should work properly, i.e., it should prevent the theme styles from being applied

### WoA sites

#### Prerequisites
- Spin up an atomic test site
- Install the [Jetpack Beta](https://github.com/Automattic/jetpack-beta/releases) plugin
- From the Jetpack Beta plugin settings page, activate this branch

#### Testing

- Install a block-based theme, such as Twenty Twenty-Four
- Notice the `Appearance` menu has no `Customize` or `Additional CSS` subitem
- Search for `custom css` in the Jetpack settings, or visit `/wp-admin/admin.php?page=jetpack#/settings?term=custom%20css`
- You should not see a `Custom CSS` section
- However, the Customizer should still be available via its URL: visit `/wp-admin/customize.php`
- Select the `Additional CSS` menu
- You should see the 3 additional options in the sidebar
- Write some custom CSS, .e.g.
```
body {
	background: purple;
}
```

- Save the changes and visit the frontend of your site
- You should see the changes you've made  (note: the Start Fresh option doesn't work for block-based themes)
- Now, install a classic theme, such as Twenty Eleven
- Repeat the operations above:
  - In the `Appearance` menu, you should see a `Customize` subitem
  - The Start Fresh option should work properly, i.e., it should prevent the theme styles from being applied

### Self-hosted sites

#### Prerequisites

- Spin up a self-hosted site with this branch
- Connect Jetpack

#### Testing

- Install a block-based theme, such as Twenty Twenty-Four
- Notice the `Appearance` menu has no `Customize` or `Additional CSS` subitem
- Search for `custom css` in the Jetpack settings, or visit `/wp-admin/admin.php?page=jetpack#/settings?term=custom%20css`
- You should not see a `Custom CSS` section
- However, the Customizer should still be available via its URL: visit `/wp-admin/customize.php`
- Select the `Additional CSS` menu
- You shouldn't see the 3 options mentioned above
- With a block-based theme, custom CSS defined in the Customizer will not be applied
- Now, install a classic theme, such as Twenty Eleven
- Repeat the operations above:
  - In the `Appearance` menu, you should see a `Customize` and `Additional CSS` subitems
- In the `Additional CSS` section of the Customizer, you still shouldn't see the 3 options mentioned above
- Write some custom CSS, .e.g.
```
body {
	background: purple;
}
```

- Save the changes and visit the frontend of your site
- You should see the changes you've made
